### PR TITLE
Ku range fix

### PR DESCRIPTION
**Pull request by kuenishi**
This is cherry-pick of #7, for https://github.com/basho/riak_cs/pull/1047. Needs tagging and webmachine update followed.

### CODE REVIEW

**Comment by shino** [comment]
Was there any conflict in cherry-pick?
I guess `range_skip_length` was independently fixed in upstream (mochi's) repository [1] and incorporated in this fork by #15 [2].

[1] https://github.com/mochi/mochiweb/commit/fe5e0c5?w=1
[2] https://github.com/basho/mochiweb/commit/b802b92262f01575536d8478ddf462cc9dedb234

**Comment by shino** [comment]
Ah, you did right.
The upstream change had a bug in case `Size < Start < End`. So, as you did, we should adopt the one in #7.

File: src/mochiweb_http.erl [hunk 1]
```diff
@@ -273,6 +271,8 @@ range_skip_length_test() ->
     %% invalid ranges
     ?assertEqual(invalid_range,
                  range_skip_length({-1, 30}, BodySize)),
+    ?assertEqual({0, BodySize},
```
Comment (shino) [inline src/mochiweb_http.erl]: The test cases around this are all for `invalid_rage` as an comment at Line 271 (new) says.

Comment (shino) [inline src/mochiweb_http.erl]: There are already cases for `Size < End` around line 258, this case should move to there.

**Comment by shino** [comment]
The test case to catch the bug when `Size < Start < End` at current master (bff08e8058bfdcfe4afa078a870df0d59ed6b09e).

```
@@ -261,6 +261,8 @@ range_skip_length_test() ->
                  range_skip_length({BodySize - 1, BodySize+5}, BodySize)),
     ?assertEqual({BodySizeLess1, 1},
                  range_skip_length({BodySize - 1, BodySize}, BodySize)),
+    ?assertEqual(invalid_range,
+                 range_skip_length({BodySize + 1, BodySize + 5}, BodySize)),
 
     %% out of range, return whole thing
```

**Comment by kuenishi** [comment]
Okay, here's updated unitests.

**Comment by shino** [comment]
Code looks nice, eunit test pass.
I will +1 after Riak CS's test passes.

**Comment by shino** [comment]
This fix works well.
+1 to merge.

**Comment by kuenishi** [comment]
Tagged v2.9.0p1 for this fix.

**Pull request merged**